src/doc/tarpl/transmutes.md

```diff
@@ -0,0 +1,29 @@
+% Transmutes
+
+Get out of our way type system! We're going to reinterpret these bits or die
+trying! Even though this book is all about doing things that are unsafe, I really
+can't emphasize that you should deeply think about finding Another Way than the
+operations covered in this section. This is really, truly, the most horribly
+unsafe thing you can do in Rust. The railguards here are dental floss.
+
+`mem::transmute<T, U>` takes a value of type `T` and reinterprets it to have
+type `U`. The only restriction is that the `T` and `U` are verified to have the
+same size. The ways to cause Undefined Behaviour with this are mind boggling.
+
+* First and foremost, creating an instance of *any* type with an invalid state
+  is going to cause arbitrary chaos that can't really be predicted.
+* Transmute has an overloaded return type. If you do not specify the return type
+  it may produce a surprising type to satisfy inference.
+* Making a primitive with an invalid value is UB
+* Transmuting between non-repr(C) types is UB
+* Transmuting an & to &mut is UB
```

Comment:
I don't believe so. `&UnsafeCell<T>` can become `&mut T` because *all* access to T must be mediated through the UnsafeCell. However `UnsafeCell<&T>` implies that T can live outside of an UnsafeCell, and therefore the compiler can't properly do the right thing with alias analysis if you were to turn that into an `&mut T`.